src/transformation/visitors/helpers/multi.ts

```diff
@@ -0,0 +1,182 @@
+    if (!ts.isCallExpression(node)) {
+        return false;
+    }
+
+    const signature = context.checker.getResolvedSignature(node);
+    return signature?.getReturnType().aliasSymbol?.declarations?.some(isMultiHelperDeclaration(context)) ?? false;
+}
+
+export function isMultiHelperNode(context: TransformationContext, node: ts.Node): boolean {
+    const type = context.checker.getTypeAtLocation(node);
+    return type.symbol?.declarations?.some(isMultiHelperDeclaration(context)) ?? false;
+}
+
+export function transformMultiHelperReturnStatement(
+    context: TransformationContext,
+    statement: ts.ReturnStatement
+): lua.Statement | undefined {
+    if (!statement.expression) return;
+    if (!ts.isCallExpression(statement.expression)) return;
+    if (!isMultiHelperCallSignature(context, statement.expression)) return;
+
+    const expressions = transformArguments(context, statement.expression.arguments);
+    return lua.createReturnStatement(expressions, statement);
+}
+
+function transformMultiHelperCallArguments(
+    context: TransformationContext,
+    expression: ts.CallExpression
+): lua.Expression[] | lua.Expression {
+    if (!isMultiHelperCallSignature(context, expression)) {
+        return context.transformExpression(expression);
+    }
+
+    if (expression.arguments.length < 1) {
+        return lua.createNilLiteral(expression);
+    }
+
+    return expression.arguments.map(e => context.transformExpression(e));
+}
+
+export function transformMultiHelperVariableDeclaration(
+    context: TransformationContext,
+    declaration: ts.VariableDeclaration
+): lua.Statement[] | undefined {
+    if (!declaration.initializer) return;
+    if (!isMultiReturnCall(context, declaration.initializer)) return;
+
+    if (!ts.isArrayBindingPattern(declaration.name)) {
+        context.diagnostics.push(invalidMultiReturnToNonArrayBindingPattern(declaration.name));
+        return [];
+    }
+
+    if (declaration.name.elements.length < 1) {
+        context.diagnostics.push(invalidMultiReturnToEmptyPatternOrArrayLiteral(declaration.name));
+        return [];
+    }
+
+    const leftIdentifiers: lua.Identifier[] = [];
+
+    for (const element of declaration.name.elements) {
+        let expression: lua.Identifier | undefined;
+
+        if (ts.isBindingElement(element) && ts.isIdentifier(element.name)) {
+            if (element.initializer) {
+                context.diagnostics.push(invalidMultiReturnArrayBindingPatternElementInitializer(element));
+                continue;
+            } else {
+                expression = transformIdentifier(context, element.name);
+            }
+        }
+
+        if (ts.isOmittedExpression(element)) {
+            expression = lua.createAnonymousIdentifier(element);
+        }
+
+        if (!expression) {
+            context.diagnostics.push(unsupportedMultiFunctionAssignment(element));
+            continue;
+        }
+
+        leftIdentifiers.push(expression);
```

Comment:
Since `ts.ArrayBindingElement` is a closed set, the only way where this condition is true is `ts.isBindingElement(element) && !ts.isIdentifier(element.name)`, which I think is cleaner to be handled explicitly.

```suggestion
        if (ts.isBindingElement(element)) {
            if (element.initializer) {
                context.diagnostics.push(invalidMultiReturnArrayBindingPatternElementInitializer(element));
            } else if (ts.isIdentifier(element.name)) {
                leftIdentifiers.push(transformIdentifier(context, element.name));
            } else {
                context.diagnostics.push(unsupportedMultiFunctionAssignment(element));
            }
        } else if (ts.isOmittedExpression(element)) {
            leftIdentifiers.push(lua.createAnonymousIdentifier(element));
        }
```